# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Zone.h
# Commit: f7b553bd9c0b
# Full Hash: f7b553bd9c0b77b0130d377f9a9c696c8c611a01
# Author: Steve Fink <sfink@mozilla.com>
# Date: 2021-05-11 21:39:06
# Regressor Bug: 1694538
# File Overlap Count: 1
# Description:
#   Bug 1694538 - Convert gcWeakKeys from <weakmap, key> tuples that require an extra lookup, to <ptr to color, target> tuples that simplify the code substantially r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106226
# ==============================================================================

diff -r 89f7028049ed -r f7b553bd9c0b js/src/gc/Zone.h
--- a/js/src/gc/Zone.h	Tue May 11 14:01:20 2021 +0000
+++ b/js/src/gc/Zone.h	Tue May 11 14:18:38 2021 +0000
@@ -266,8 +266,8 @@
   // Mapping from not yet marked keys to a vector of all values that the key
   // maps to in any live weak map. Separate tables for nursery and tenured
   // keys.
-  js::ZoneOrGCTaskData<js::gc::WeakKeyTable> gcWeakKeys_;
-  js::ZoneOrGCTaskData<js::gc::WeakKeyTable> gcNurseryWeakKeys_;
+  js::ZoneOrGCTaskData<js::gc::EphemeronEdgeTable> gcEphemeronEdges_;
+  js::ZoneOrGCTaskData<js::gc::EphemeronEdgeTable> gcNurseryEphemeronEdges_;
 
   // Keep track of all RttValue and related objects in this compartment.
   // This is used by the GC to trace them all first when compacting, since the
@@ -539,18 +539,21 @@
 
   void beforeClearDelegateInternal(JSObject* wrapper, JSObject* delegate);
   void afterAddDelegateInternal(JSObject* wrapper);
-  js::gc::WeakKeyTable& gcWeakKeys() { return gcWeakKeys_.ref(); }
-  js::gc::WeakKeyTable& gcNurseryWeakKeys() { return gcNurseryWeakKeys_.ref(); }
+  js::gc::EphemeronEdgeTable& gcEphemeronEdges() {
+    return gcEphemeronEdges_.ref();
+  }
+  js::gc::EphemeronEdgeTable& gcNurseryEphemeronEdges() {
+    return gcNurseryEphemeronEdges_.ref();
+  }
 
-  js::gc::WeakKeyTable& gcWeakKeys(const js::gc::Cell* cell) {
-    return cell->isTenured() ? gcWeakKeys() : gcNurseryWeakKeys();
+  js::gc::EphemeronEdgeTable& gcEphemeronEdges(const js::gc::Cell* cell) {
+    return cell->isTenured() ? gcEphemeronEdges() : gcNurseryEphemeronEdges();
   }
 
   // Perform all pending weakmap entry marking for this zone after
   // transitioning to weak marking mode.
   js::gc::IncrementalProgress enterWeakMarkingMode(js::GCMarker* marker,
                                                    js::SliceBudget& budget);
-  void checkWeakMarkingMode();
 
   // A set of edges from this zone to other zones used during GC to calculate
   // sweep groups.
@@ -675,7 +678,7 @@
 
   bool isQueuedForBackgroundSweep() { return isOnList(); }
 
-  void sweepWeakKeysAfterMinorGC();
+  void sweepEphemeronTablesAfterMinorGC();
 
   FinalizationRegistrySet& finalizationRegistries() {
     return finalizationRegistries_.ref();